jax/_src/interpreters/mlir.py

```diff
@@ -1502,9 +1502,12 @@ def lower_multi_platform(ctx: LoweringRuleContext,
    rule_args: the args of the lowering rules.
    rule_kwargs: the kwargs of the lowering rules.
   """
-  assert isinstance(ctx.module_context.lowering_parameters.platforms, tuple)
-  platforms = ctx.module_context.lowering_parameters.platforms
-  platforms_with_specific_rules = util.flatten(
+  platforms: Sequence[str]
+  if ctx.module_context.lowering_parameters.is_multi_platform:
+    platforms = ctx.module_context.lowering_parameters.platforms  # type: ignore
```

Comment:
Why the type: ignore?

If it is because `platforms` is optional in the type of that structure, then you should use `assert platforms is not None` instead.

Reply:
I did not know that mypy understands this. Thanks!